### PR TITLE
Pull Request for Issue2243: adding control move progress to control editors

### PR DESCRIPTION
**Pull request by helfrij**
- Created new class BioXASValueProgressLabel. This class inherits from QLabel with a couple of class variables related to displaying move progress: progressMin, progressValue, progressMax, displayProgress.
- Added BioXASValueProgressLabel to BioXASValueEditor. Added class variables for the progress values, with getters, setters, and signals.
- Added methods to BioXASControlEditor that update the progress min, max, value, and display flag in response to changes in the move status of the control.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review?

File: source/ui/BioXAS/BioXASValueProgressLabel.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	painter.drawText(labelRect, Qt::AlignCenter, text());
+
+	painter.end();
+}
+
+double BioXASValueProgressLabel::progressPercentage(double min, double value, double max) const
+{
+	return abs(value - min) / abs(max - min);
```
Comment (dretrex) [inline source/ui/BioXAS/BioXASValueProgressLabel.cpp]: For information only! You could use qAbs() which would technically give you a platform independent abs function.

Comment (dretrex) [inline source/ui/BioXAS/BioXASValueProgressLabel.cpp]: Oh! You implemented your own abs.  I'd recommend just using qAbs().  Way simpler.

File: source/ui/BioXAS/BioXASControlEditor.cpp [hunk 2]
```diff
@@ -262,6 +291,32 @@ void BioXASControlEditor::updateUnits()
...
+void BioXASControlEditor::updateProgressValueMaximum()
+{
+	if (control_ && useControlMovingAsProgress_ && control_->moveInProgress())
+		BioXASValueEditor::setProgressValueMaximum(control_->setpoint());
+}
+
+void BioXASControlEditor::updateProgressValue()
+{
+	if (control_ && useControlMovingAsProgress_ && control_->moveInProgress())
+		BioXASValueEditor::setProgressValue(control_->value());
```
Comment (dretrex) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: Why do you need to specifically scope the BioXASValueEditor?  I didn't notice any inheritance or conflicts in your changes.

**Comment by dretrex** [comment]
There are merge conflicts in this guy after all the merging I did.  Looks good other than a couple small comments.

Time Estimate: 5 minutes

File: source/ui/BioXAS/BioXASValueProgressLabel.cpp (hunk 1 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASValueProgressLabel.cpp]: Kk! I'll make the change.

File: source/ui/BioXAS/BioXASControlEditor.cpp (hunk 2 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: There's no real need to scope it here. But the majority of methods are reimplemented between BioXASControlEditor and BioXASValueEditor. In this case, I think, it helps to be specific.

**Comment by helfrij** [comment]
Thanks @dretrex. Merge conflicts should be resolved.

File: source/ui/BioXAS/BioXASControlEditor.cpp (hunk 2 above)
Comment (dretrex) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: Ah, okay.

**Pull request merged**